aspnetcore/fundamentals/file-providers.md

```diff
@@ -1,148 +1,261 @@
 
-**`*`**
+In the sample app, the *WatchConsole* console app is configured to display a message whenever a text file is modified:
 
-   Matches anything at the current folder level, or any filename, or any file extension. Matches are terminated by `/` and `.` characters in the file path.
+::: moniker range=">= aspnetcore-2.0"
 
-<strong><code>**</code></strong>
+[!code-csharp[](file-providers/samples/2.x/WatchConsole/Program.cs?name=snippet1&highlight=1-2,16,19-20)]
 
-   Matches anything across multiple directory levels. Can be used to recursively match many files within a directory hierarchy.
+::: moniker-end
 
-### Globbing pattern examples
+::: moniker range="< aspnetcore-2.0"
 
-**`directory/file.txt`**
+[!code-csharp[](file-providers/samples/1.x/WatchConsole/Program.cs?name=snippet1&highlight=1-2,16,19-20)]
 
-   Matches a specific file in a specific directory.
+::: moniker-end
 
-**<code>directory/*.txt</code>**
+Some file systems, such as Docker containers and network shares, may not reliably send change notifications. Set the `DOTNET_USE_POLLINGFILEWATCHER` environment variable to `1` or `true` to poll the file system for changes every four seconds.
 
-   Matches all files with `.txt` extension in a specific directory.
+## Glob patterns
 
-**`directory/*/bower.json`**
+File system paths use wildcard patterns called *glob (or globbing) patterns*. Specify groups of files with these patterns. The two wildcard characters are `*` and `**`:
 
-   Matches all `bower.json` files in directories exactly one level below the `directory` directory.
+**`*`**  
+Matches anything at the current folder level, any filename, or any file extension. Matches are terminated by `/` and `.` characters in the file path.
 
-**<code>directory/&#42;&#42;/&#42;.txt</code>**
+**`**`**  
+Matches anything across multiple directory levels. Can be used to recursively match many files within a directory hierarchy.
 
-   Matches all files with `.txt` extension found anywhere under the `directory` directory.
+**Glob pattern examples**
 
-## File Provider usage in ASP.NET Core
+**`directory/file.txt`**  
+Matches a specific file in a specific directory.
 
-Several parts of ASP.NET Core utilize file providers. `IHostingEnvironment` exposes the app's content root and web root as `IFileProvider` types. The static files middleware uses file providers to locate static files. Razor makes heavy use of `IFileProvider` in locating views. Dotnet's publish functionality uses file providers and globbing patterns to specify which files should be published.
+**`directory/*.txt`**  
+Matches all files with `.txt` extension in a specific directory.
 
-## Recommendations for use in apps
+**`directory/*/appsettings.json`**  
+Matches all `appsettings.json` files in directories exactly one level below the `directory` folder.
```

Comment:
Italicize directory names

Reply:
> Italicize directory names

This time, it's a placeholder. I'll rework this one, too, and see if I can improve it.